src/prefect/transactions.py

```diff
@@ -218,6 +234,35 @@ def transaction(
     commit_mode: CommitMode = CommitMode.LAZY,
     overwrite: bool = False,
 ) -> Generator[Transaction, None, None]:
+    """
+    A context manager for opening and managing a transaction.
+
+    Args:
+        - key: An identifier to use for the transaction
+        - store: The store to use for persisting the transaction result
+        - commit_mode: The commit mode controlling when the transaction and
+            child transactions are committed
+        - overwrite: Whether to overwrite an existing transaction record in the store
+
+    Yields:
+        - Transaction: An object representing the transaction state
+    """
+    if not store and (flow_run_context := FlowRunContext.get()):
+        if flow_run_context:
+            new_factory = flow_run_context.result_factory.model_copy(
+                update={
+                    "persist_result": True,
+                }
+            )
+            if not new_factory.storage_block:
+                default_storage = get_default_result_storage(_sync=True)
+                new_factory.storage_block = default_storage
+            store = ResultFactoryStore(
+                result_factory=new_factory,
+            )
+
+    if not key:
+        key = str(uuid.uuid4())
```

Comment:
This unfortunately won't work in certain situations - the one that immediately comes to mind is if a task has a `NONE` cache policy; is there a reason you needed this default?  The task engine should always be providing a relevant key

Reply:
I added this default so that using the `transaction` context manager directly would persist a record by default, but I realize now that doesn't make sense. I'll remove this.